packages/medusa/src/utils/decorators/atomic-phase.ts

```diff
@@ -0,0 +1,123 @@
+import { TransactionBaseService } from "../../interfaces";
+import { EntityManager } from "typeorm";
+import { IsolationLevel } from "typeorm/driver/types/IsolationLevel";
+import { formatException } from "../exception-formatter";
+
+export type CustomAtomicPhaseExceptionHandler = <T>(this: T, error: unknown) => unknown | Promise<unknown>
+
+export function UseAtomicPhase(customAtomicPhaseExceptionHandler?: CustomAtomicPhaseExceptionHandler): MethodDecorator {
```

Comment:
Todo: export from the utils index
Todo: rename file to use-atomic-phase

Reply:
Todo: return type of the type should be never | Promise<never>

Reply:
suggestion: would drop the `Use` and rename the decorator to be:
```
@AtomicPhase()
...
```